torch/ao/quantization/_pt2e/prepare.py

```diff
@@ -22,9 +22,14 @@ def _maybe_insert_input_and_output_observers_for_node(
 ):
     this_node_dtype_info = node.meta["target_dtype_info"] if "target_dtype_info" in node.meta else None
     if "val" in node.meta:
+        def check_node_meta_val_fake_tensor(node_meta_val):
+            if isinstance(node_meta_val, (tuple, list)):
+                return all(check_node_meta_val_fake_tensor(element) for element in node_meta_val)
+            else:
+                return isinstance(node_meta_val, FakeTensor)
         output_is_a_tensor = (
             this_node_dtype_info is not None and
-            isinstance(node.meta["val"], FakeTensor)
+            check_node_meta_val_fake_tensor(node.meta["val"])
```

Comment:
not this one, L41 should be applied after L56

Reply:
Looks like we can't simply move L41 after L56 due to the case https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/_pt2e/prepare.py#L30
When the node has no `target_dtype_info`, we should also skip the insert of input observer. Does my understanding right?

Reply:
Move L41 after L56 and also add `skip_inserting_input_observers` to skip the insert of input observer when `node.meta["target_dtype_info"]` not set for this node. @jerryzh168 Please help to take a look of this PR again.